hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -457,6 +461,37 @@ private void saveWorkloadProfileMetadataToInflight(WorkloadProfile profile, Hood
   private JavaRDD<WriteStatus> upsertRecordsInternal(JavaRDD<HoodieRecord<T>> preppedRecords, String commitTime,
       HoodieTable<T> hoodieTable, final boolean isUpsert) {
 
+    if (getConfig().getSchemaCheck()) {
```

Comment:
this whole block can be pulled into a private method like `validateSchema()` and reused? without being inlined into `upsertRecordsInternal()`

Reply:
done.